pkg/leeway/build.go

```diff
@@ -2645,3 +2653,99 @@ func isEmpty(dir string) bool {
 	}
 	return len(entries) == 0
 }
+
+// sortPackagesByDependencyDepth sorts packages by their dependency depth (deepest first).
+// This prioritizes downloading packages on the critical path, allowing dependent builds
+// to start earlier and reducing overall wall-clock build time.
+//
+// Algorithm:
+// 1. Calculate dependency depth for each package (max distance from leaf nodes)
+// 2. Sort packages by depth in descending order (deepest = most dependencies = critical path)
+// 3. Packages with equal depth maintain their relative order (stable sort)
+func sortPackagesByDependencyDepth(packages []*Package) []*Package {
+	if len(packages) <= 1 {
+		return packages
+	}
+
+	// Calculate dependency depth for each package
+	depthCache := make(map[string]int)
+	for _, pkg := range packages {
+		calculateDependencyDepth(pkg, depthCache)
+	}
+
+	// Create a copy to avoid modifying the input slice
+	sorted := make([]*Package, len(packages))
+	copy(sorted, packages)
+
+	// Sort by depth (descending) - packages with more dependencies first
+	// This is a stable sort, so packages with equal depth maintain their order
```

Comment:
Is this statement true? Is this stable? 🤔 I think the implemented algorithm is NOT a stable sort. A stable sort preserves the relative order of equal elements. The nested loop swap pattern used here does not guarantee stability, does it? The test `TestSortPackagesByDependencyDepth_Stability` passes by coincidence because the input is already in the desired order.